src/util/play_time.rs

```diff
@@ -0,0 +1,165 @@
+    }
+}
+
+// REMOVED: Copy trait (SystemTime is not Copy)
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct PlayTimeState {
+    #[serde(with = "serial_dur")]
+    pub total_active_duration: Duration,
+
+    #[serde(skip)]
+    pub current_session_start: Option<Instant>,
+
+    #[serde(with = "serial_secs")]
+    pub last_activity_time: SystemTime,
+
+    // CHANGED: Points to specific default function
+    #[serde(default = "serial_secs::default_time", with = "serial_secs")]
+    pub last_break_end: SystemTime,
+}
+
+impl PlayTimeState {
+    pub fn new() -> Self {
+        Self {
+            total_active_duration: Duration::ZERO,
+            current_session_start: None,
+            last_activity_time: SystemTime::now(),
+            last_break_end: SystemTime::now(),
+        }
+    }
+
+    pub fn get_total_active_time(&self) -> Duration {
+        self.total_active_duration
+            + self
+                .current_session_start
+                .map_or(Duration::ZERO, |s| s.elapsed())
+    }
+
+    pub fn is_active(&self) -> bool {
+        self.current_session_start.is_some()
+    }
+
+    pub fn start_session(&mut self) {
+        if self.current_session_start.is_none() {
+            self.current_session_start = Some(Instant::now());
+            // Also update activity time immediately
+            self.last_activity_time = SystemTime::now();
+        }
+    }
+
+    /// Called periodically while recording is active to mark ongoing activity
+    pub fn tick_activity(&mut self) {
+        // Keep updating the "Last Seen" time to Now while playing.
+        // This ensures that when the recording stops, this timestamp
+        // marks the exact end of the session.
+        self.last_activity_time = SystemTime::now();
+    }
+
+    pub fn pause_session(&mut self) {
+        if let Some(start) = self.current_session_start.take() {
+            self.total_active_duration += start.elapsed();
+        }
+        // Ensure we mark the end of the session
+        self.last_activity_time = SystemTime::now();
+    }
+
+    /// Called ONLY when the Recorder stops specifically due to IDLE
+    pub fn cancel_idle_buffer(&mut self) {
+        // We subtract the idle duration (e.g. 30s) from the total
+        // because the Recorder includes it before triggering the stop.
+        let idle_buffer = constants::MAX_IDLE_DURATION;
+        if self.total_active_duration >= idle_buffer {
+            self.total_active_duration -= idle_buffer;
+        }
+    }
+
+    pub fn reset(&mut self) {
+        *self = Self::new();
+    }
+
+    pub fn should_reset(&self) -> bool {
+        let idle = self.last_activity_time.elapsed().unwrap_or_default();
+        let since_break = self.last_break_end.elapsed().unwrap_or_default();
+
+        idle > constants::PLAY_TIME_BREAK_THRESHOLD
+            || since_break > constants::PLAY_TIME_ROLLING_WINDOW
+    }
+
+    pub fn save(&self) -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
use `color_eyre::eyre::Result` (importing it for the module) instead of a `Result` with a `Box<dyn Error>`; the former offers similar type erasure, but with better callstacks + colours + traceability